src/main/java/core/basesyntax/model/FigureSupplier.java

```diff
@@ -0,0 +1,42 @@
+package core.basesyntax.model;
+
+import java.util.Random;
+
+public class FigureSupplier {
+    private static final int FIGURE_COUNT = 5;
+    private static final int SIZE_OF_UNITS = 50;
+    private Random random = new Random();
+    private ColorSupplier colorSupplier = new ColorSupplier();
+    private Figure figure;
+
+    public Figure getRandomFigure() {
+        int figureNumber = random.nextInt(FIGURE_COUNT);
+        switch (figureNumber) {
+            case 0:
+                figure = new Circle(colorSupplier.getRandomColor(),
+                        new Random().nextInt(SIZE_OF_UNITS) + 1);
+                break;
+            case 1:
+                figure = new IsoscelesTrapezoid(colorSupplier.getRandomColor(),
+                        random.nextInt(SIZE_OF_UNITS),
+                        random.nextInt(SIZE_OF_UNITS),
+                        random.nextInt(SIZE_OF_UNITS));
```

Comment:
don't pass expressions in constructor arguments. let's create separate variables and pass them.  and to keep code clean do all of these in a private method.